PocketKit/Sources/PocketKit/PocketAppDelegate.swift

```diff
@@ -80,6 +80,30 @@ public class PocketAppDelegate: UIResponder, UIApplicationDelegate {
         Textiles.initialize()
         refreshCoordinator.initialize()
 
+        let legacyUserMigration = LegacyUserMigration(
+            userDefaults: userDefaults,
+            encryptedStore: PocketEncryptedStore(),
+            appSession: appSession
+        )
+
+        do {
+            let attempted = try legacyUserMigration.perform()
+            if attempted {
+                Crashlogger.breadcrumb(category: "launch", level: .info, message: "Legacy user migration required; running.")
+            } else {
+                Crashlogger.breadcrumb(category: "launch", level: .info, message: "Legacy user migration not required; skipped.")
+            }
+        } catch LegacyUserMigrationError.missingStore {
+            Crashlogger.breadcrumb(category: "launch", level: .info, message: "No previous store for user migration; skipped.")
+            // Since we don't have a store, we can skip any further attempts at running this migration.
+            legacyUserMigration.forceSkip()
+        } catch {
+            // All errors are something we can't resolve client-side, so we don't want to re-attempt
+            // on further launches.
+            legacyUserMigration.forceSkip()
+            Crashlogger.capture(error: error)
+        }
```

Comment:
@bassrock This chunk of code is the same for the main app and the save extension, but I want to make sure of something. This is the flow based on this implementation as-is.

1. Attempt to perform the migration. If it succeeds, continue as normal.
2. Attempt to perform the migration. If there is no store, mark the migration as complete, and continue as normal (the user will be logged out). This is the flow new users or fresh installs will see.
3. Attempt to perform the migration. If there is no store, mark the migration as complete (since we can't resolve this client-site (failed to decrypt, for example)), and continue with the user logged out.

Situation 3 is the one I'm not happiest about, but I believe that it has to be done this way.

Any thoughts?